plugins/database/postgresql/postgresql.go

```diff
@@ -206,6 +206,22 @@ func (p *PostgreSQL) CreateUser(ctx context.Context, statements dbplugin.Stateme
 
 	// Execute each query
 	for _, stmt := range statements.Creation {
+		// A postgres query may have multi lines, and if so, it will include the word
+		// "END". For example, this statement should be run as one transaction:
+		// "DO $$ BEGIN IF NOT EXISTS (SELECT FROM pg_catalog.pg_roles WHERE rolname='my_role') THEN CREATE ROLE my_role; END IF; END $$"
+		if strings.Contains(stmt, "END") {
```

Comment:
What if the string "END" occurs naturally within the string, e.g. `APPEND`?  I feel like to make this safe we'd need to at least match only exact words, and ideally exclude anything within quoted delimiters.

Reply:
Ah, great point. I will change that.

Reply:
This feedback is incorporated in https://github.com/hashicorp/vault/pull/8512/commits/c80d37c5ae9d840d2c1f9704c8132f52defefb19 and test cases are added to verify the underlying logic.